File: SDKs/amazonaws.com/lakeformation/2017-03-31/typescript/src/sdk/models/operations/putdatalakesettings.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum PutDataLakeSettingsXAmzTargetEnum {
    AwsLakeFormationPutDataLakeSettings = "AWSLakeFormation.PutDataLakeSettings"
}


export class PutDataLakeSettingsHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: PutDataLakeSettingsXAmzTargetEnum;
}


export class PutDataLakeSettingsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: PutDataLakeSettingsHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.PutDataLakeSettingsRequest;
}


export class PutDataLakeSettingsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  internalServiceException?: any;

  @SpeakeasyMetadata()
  invalidInputException?: any;

  @SpeakeasyMetadata()
  putDataLakeSettingsResponse?: Map<string, any>;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/amazonaws.com/lakeformation/2017-03-31/python/src/sdk/models/operations/putdatalakesettings.py
from dataclasses import dataclass, field
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Any,Optional
from enum import Enum
from sdk.models import shared

class PutDataLakeSettingsXAmzTargetEnum(str, Enum):
    AWS_LAKE_FORMATION_PUT_DATA_LAKE_SETTINGS = "AWSLakeFormation.PutDataLakeSettings"


@dataclass
class PutDataLakeSettingsHeaders:
    x_amz_target: PutDataLakeSettingsXAmzTargetEnum = field(metadata={'header': { 'field_name': 'X-Amz-Target', 'style': 'simple', 'explode': False }})
    x_amz_algorithm: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Algorithm', 'style': 'simple', 'explode': False }})
    x_amz_content_sha256: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Content-Sha256', 'style': 'simple', 'explode': False }})
    x_amz_credential: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Credential', 'style': 'simple', 'explode': False }})
    x_amz_date: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Date', 'style': 'simple', 'explode': False }})
    x_amz_security_token: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Security-Token', 'style': 'simple', 'explode': False }})
    x_amz_signature: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-Signature', 'style': 'simple', 'explode': False }})
    x_amz_signed_headers: Optional[str] = field(default=None, metadata={'header': { 'field_name': 'X-Amz-SignedHeaders', 'style': 'simple', 'explode': False }})
    

@dataclass
class PutDataLakeSettingsRequest:
    headers: PutDataLakeSettingsHeaders = field()
    request: shared.PutDataLakeSettingsRequest = field(metadata={'request': { 'media_type': 'application/json' }})
    

@dataclass
class PutDataLakeSettingsResponse:
    content_type: str = field()
    status_code: int = field()
    internal_service_exception: Optional[Any] = field(default=None)
    invalid_input_exception: Optional[Any] = field(default=None)
    put_data_lake_settings_response: Optional[dict[str, Any]] = field(default=None)
    
